packages/api/src/modules/providers/chorus/repositories/chorus.line.repository.test.ts

```diff
@@ -0,0 +1,39 @@
+import { MongoServerError } from "mongodb";
+import chorusLineRepository from "./chorus.line.repository";
+import MongoRepository from "../../../../shared/MongoRepository";
+import * as MongoHelper from "../../../../shared/helpers/MongoHelper";
+const mockedMongoHelper = jest.mocked(MongoHelper);
+jest.mock("../../../../shared/helpers/MongoHelper");
+describe("ChorusLineRepository", () => {
+    const MONGO_SERVER_ERROR = new MongoServerError({ message: "Duplicate Error" });
+    MONGO_SERVER_ERROR.code = 11000;
+    MONGO_SERVER_ERROR.writeErrors = [];
+
+    let mockInsertMany = jest.fn().mockRejectedValue(MONGO_SERVER_ERROR);
+
+    let mockMongoRepositoryCollection: jest.SpyInstance;
+    beforeAll(() => {
+        mockMongoRepositoryCollection = jest
+            // @ts-expect-error: test
+            .spyOn(MongoRepository.prototype, "collection", "get")
+            // @ts-expect-error: test
+            .mockReturnValue({ insertMany: mockInsertMany });
+    });
+
+    afterEach(() => mockInsertMany.mockReset());
+
+    describe("insertMany", () => {
+        beforeEach(() => {
+            mockedMongoHelper.isDuplicateError.mockReturnValue(true);
+        });
+
+        it("should call buildDuplicateIndexError", async () => {
+            try {
+                await chorusLineRepository.insertMany([]);
+                throw new Error("Error should have been throwned");
+            } catch (e) {
+                expect(mockedMongoHelper.buildDuplicateIndexError).toHaveBeenCalledWith(MONGO_SERVER_ERROR);
+            }
```

Comment:
Je trouve pas clair cette tournure de code. Ça paraît bizarre que tu lèves une exception comme ça (même si je finis par comprendre que c'est pour être sûr d'entrer dans le `catch`)
```suggestion
            try {
                await chorusLineRepository.insertMany([]);
            } finally {
                            expect(mockedMongoHelper.buildDuplicateIndexError).toHaveBeenCalledWith(MONGO_SERVER_ERROR);
            }
```
?

Reply:
ah mais oui voilà ce que je cherchais ! j'avais oublié ce bon vieux `finally`. Merci, je change ça demain

Reply:
Ne fonctionne pas car il faut quand même un catch pour rattraper l'erreur. Je fais un mix